### PR TITLE
Extend nGraph for operation SoftSign-9

### PR DESCRIPTION
**Pull request by yekruglov**
Tickets: CVS-79216, CVS-79217, CVS-79227

Changes:
* nGraph shell implementation for SoftSign-9
* nGraph reference implementation for SoftSign-9
* Decomposition transformation. It is disabled for GNA plugin since SoftSign is supported by GNA.

### CODE REVIEW

File: src/core/src/op/softsign.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    return rc;
+}
+
+}  // namespace
+
+// *** SOFTSIGN OP SET V9 **
+BWDCMP_RTTI_DEFINITION(ov::op::v9::SoftSign);
+
+ov::op::v9::SoftSign::SoftSign(const Output<Node>& arg) : Op({arg}) {
```
Comment (rkazants) [inline src/core/src/op/softsign.cpp:44]: please inherit from base class for all activation operations like `UnaryElementwiseArithmetic`

File: src/core/src/op/softsign.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    NGRAPH_OP_SCOPE(v9_SoftSign_validate_and_infer_types);
+    const element::Type& input_et = get_input_element_type(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          input_et.is_dynamic() || input_et.is_real(),
+                          "Input element type must be float. Got: ",
+                          input_et);
+
+    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+}
```
Comment (rkazants) [inline src/core/src/op/softsign.cpp:63]: not needed if you inherit from `UnaryElementwiseArithmetic`

File: src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/single_op_graph.cpp [hunk 3]
```diff
@@ -993,6 +993,13 @@ std::shared_ptr<ov::Model> generate(const std::shared_ptr<ov::op::v4::SoftPlus>
     return std::make_shared<ov::Model>(results, params, "SoftPlusGraph");
 }
 
+std::shared_ptr<ov::Model> generate(const std::shared_ptr<ov::op::v9::SoftSign> &node) {
+    const auto params = ngraph::builder::makeDynamicParams(ov::element::f16, {{4, 4}});
+    auto Node = std::make_shared<ov::op::v9::SoftSign>(params.at(0));
+    ov::ResultVector results{std::make_shared<ov::op::v0::Result>(Node)};
+    return std::make_shared<ov::Model>(results, params, "SoftSignGraph");
+}
+
```
Comment (rkazants) [inline src/tests/functional/plugin/shared/src/single_layer_tests/op_impl_check/single_op_graph.cpp:1002]: please add test for reference implementation

File: src/core/include/ngraph/op/softsign.hpp [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
```
Comment (ilyachur) [inline src/core/include/ngraph/op/softsign.hpp:7]:
```suggestion
```

File: src/core/include/openvino/op/softsign.hpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+#include "openvino/op/op.hpp"
+
+namespace ov {
+namespace op {
+namespace v9 {
+class OPENVINO_API SoftSign : public Op {
+public:
+    OPENVINO_OP("SoftSign", "opset9");
+    BWDCMP_RTTI_DECLARATION;
```
Comment (ilyachur) [inline src/core/include/openvino/op/softsign.hpp:15]:
```suggestion
```

File: src/core/include/openvino/op/softsign.hpp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    /// Output `[d0, ...]`
+    ///
+    SoftSign(const Output<Node>& arg);
+
+    bool visit_attributes(AttributeVisitor& visitor) override;
+    void validate_and_infer_types() override;
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override;
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    bool evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const override;
```
Comment (ilyachur) [inline src/core/include/openvino/op/softsign.hpp:31]: Why do we need evaluate for this operation?

Also please use evaluate with ov::Tensor.

File: src/core/src/op/softsign.cpp [hunk 7]
```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/op/softsign.hpp"
```
Comment (ilyachur) [inline src/core/src/op/softsign.cpp:5]:
```suggestion
#include "openvino/op/softsign.hpp"
```

File: src/core/src/op/softsign.cpp [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+#include <ngraph/validation_util.hpp>
+
+#include "itt.hpp"
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/runtime/reference/softsign.hpp"
```
Comment (ilyachur) [inline src/core/src/op/softsign.cpp:13]: Remove redundant headers and use headers from openvino folder

File: src/core/src/op/softsign.cpp (hunk 1 above)
Comment (ilyachur) [reply src/core/src/op/softsign.cpp:44]: In this case you need to have BWD macros. Please don't remove it if you change base class

**Comment by andrei-kochin** [comment]
@gkrivor please take a look at op-check part

File: src/core/src/op/softsign.cpp [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    default:
+        rc = false;
+        break;
+    }
+    return rc;
+}
+
+}  // namespace
+
+// *** SOFTSIGN OP SET V9 **
```
Comment (rkazants) [inline src/core/src/op/softsign.cpp:38]: nit: please remove this useless comment)

**Review by rkazants — APPROVED** [review]
Looks good to me

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**